checks/ip/sap.rb

```diff
@@ -0,0 +1,28 @@
+module Intrigue
+  module Ident
+    module IpCheck
+      class SAP < Intrigue::Ident::IpCheck::Base
+        def generate_checks
+          [
+            {
+              type: 'fingerprint',
+              category: 'application',
+              tags: %w[Network Tcp SAP],
+              vendor: 'SAP',
+              product: 'SAProuter',
+              description: 'match via protocol response string',
+              references: [
+                'https://support.sap.com/en/tools/connectivity-tools/saprouter.html'
+              ],
+              request_type: :plain,
+              protocol: :tcp,
+              request_content: "WHOAREYOU?",
```

Comment:
lol? is this part of the spec or do you need to just send a string?

Reply:
In the nuclei template they make the request with hex data: https://github.com/projectdiscovery/nuclei-templates/blob/master/network/sap-router.yaml

I just reversed the hex data to ASCII: 
![image](https://user-images.githubusercontent.com/52205434/123560050-4fe3cb00-d7a0-11eb-8e29-1f415cdb9592.png)

I did that because in the IP checks in Ident if we make a `hex` request type (instead of `plain`) then both the input and the output will be hex data:

https://github.com/intrigueio/intrigue-ident/blob/31af0b96e4dbe69877ebe25f0a40f51a3e9e3bf0/lib/ip/ip.rb#L55

But in this case we get a string response back, and we want to catch that. Of course we can modify the above hex sending method to account for these special cases, but it was just easier to convert the hex request content to ASCII.

Reply:
The result is the same, I tested both ways (request data being hex or string).